src/current/v23.1/cdc-queries.md

```diff
@@ -190,6 +190,59 @@ For newly inserted rows in a table, the `cdc_prev` column will emit as `NULL`.
+
+If you have TTL logic defined with [`ttl_expiration_expression` or `ttl_expire_after`]({% link {{ page.version.version }}/row-level-ttl.md %}#syntax-overview), you can leverage CDC queries to determine whether or not a given row was expired at the time of the changefeed event, including a delete event.
+
+To emit rows that were deleted after expiring, you can use syntax similar to the following:
+
+{% include_cached copy-clipboard.html %}
+~~~sql
+CREATE CHANGEFEED INTO 'external://sink'
+AS SELECT cdc_prev FROM test_table
+WHERE event_op() = 'delete'
+AND (cdc_prev).crdb_internal_expiration < statement_timestamp();
+~~~
+
+This query:
+
+- Accesses the `cdc_prev` column for the previous state of the row.
+- Searches for `delete` events in that previous state.
+- Finds the TTL expiration timestamp of the deleted rows where it is earlier than the current statement timestamp.
+
+{{site.data.alerts.callout_info}}
+This will only emit rows that were deleted **after** expiring. Furthermore, consider that a transactional SQL delete during the window between the row expiring and the TTL job running will also cause this message to emit from the changefeed.
+{{site.data.alerts.end}}
+
+Equally, you can filter out delete messages for expired rows from your changefeed:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE CHANGEFEED AS SELECT cdc_prev FROM test_table
+WHERE NOT (event_op() = 'delete'
+AND (cdc_prev).crdb_internal_expiration < statement_timestamp());
+~~~
+
+In some cases, you may have custom expiration logic on rows in a table. Custom TTL expressions and CDC queries evaluate in different ways.
+
+- Custom TTL expressions evaluate to a timestamp after which the row expires. For example, if you want to configure a table so that some rows never expire, your expression needs to evaluate to `'infinity'` for those rows.
+- CDC queries filter using `WHERE` clauses, which are booleans.
+
+To preserve your custom TTL expression in a CDC query, you can convert the TTL expression to a boolean by wrapping it in a parenthesis and appending `< statement_timestamp()`.
+
+In the following example, the CDC query includes a custom TTL expression that is evaluated against the statement timestamp. In addition to the original TTL expression, it also accesses the `cdc_prev` column in order to evaluate the previous state of the rows:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE CHANGEFEED AS SELECT cdc_prev FROM ttl_test_per_row
+WHERE NOT (event_op() = 'delete'
+AND (CASE WHEN (cdc_prev).immortal
+THEN 'infinity'
+ELSE (cdc_prev).expired_at END) < statement_timestamp());
```

Comment:
we really need to manually verify that these examples work.
Do we know if these examples work?

Reply:
@miretskiy So, top two examples, all good 👍 work fine

The final example regarding custom TTL expression — I don't think we should include my example. I was messing around with the logic, but I don't think `infinity` will play well in CRL having now read the timestamp docs. 

This example is tricky given that this is more about a customer's _custom_ logic. What do you think about signposting as how you would write the syntax, and provide some (kind of) pseudo SQL to describe, something like:

```
CREATE CHANGEFEED AS SELECT cdc_prev FROM {table} WHERE NOT ( event_op() = 'delete' AND your_custom_ttl_expression_for_the_row() < statement_timestamp() );
```

I'd have to note this very clearly. What do you think? Or, do you think there is a better way to represent the custom logic?

Reply:
Infinity would not work, but maybe explicitly giving ttl expression example (like we have on TTL docs page):
```
CREATE TABLE ttl_test_per_row (
  id UUID PRIMARY KEY default gen_random_uuid(),
  description TEXT,
  expired_at TIMESTAMPTZ
) WITH (ttl_expiration_expression = 'expired_at');
```

And then just referencing expired_at column in the cdc query?